### PR TITLE
refactor: modify worker file path to satisfy the dependency analysis

### LINKED ISSUE #71: Missing worker file after build with nuxt3
## Description

I was using tinypool in my nuxt3 project, but after build I found that some of tinypool's dependencies would be lost(not included in the `.output`). After checking, I found that the [nitro](https://github.com/unjs/nitro) packager used by nuxt3 relies on [@vercel/nft](https://github.com/vercel/nft) for dependency analysis. Due to the way tinypool writes the worker file, they are not collected by the dependency and are missing in the build result.

https://github.com/tinylibs/tinypool/blob/a5b666924a6dda428d38341d79404c953b2f2b8a/src/runtime/thread-worker.ts#L13-L15

Compare this with the way piscina is written which can be collected by the @vercel/nft, the related worker files will not be missing after build.

https://github.com/piscinajs/piscina/blob/a2cf1197cfc81566b4e5028ca3ea96128834ddff/src/index.ts#L590

## Reproduce

For a comparison of the dependency analysis results, see this [reproduce](https://stackblitz.com/edit/stackblitz-starters-mcb5v4?file=test%2Ftinypool.test.ts&file=test%2Fpiscina.test.ts)

- `npm run test`, you will see the difference
image

## Suggestion

Since @vercel/nft's analysis logic has compatible processing for `__dirname` and `fileURLToPath(import.meta.url)`.
So can we change the way the worker file is written so that similar build tools can correctly analyze all dependencies without breaking the code logic? Maybe like this ↓

```diff
- const __dirname = dirname(fileURLToPath(import.meta.url))
- this.thread = new Worker(resolve(__dirname, './entry/worker.js'), options)
+ this.thread = new Worker(fileURLToPath(import.meta.url + '/../entry/worker.js'), options);
```

(Willing to contribute a PR for this 🙌🏻)

### PR DESCRIPTION
**Pull request by kricsleo**
close #71

### CODE REVIEW

**Review by AriPerkkio — CHANGES_REQUESTED** [review]
Same change should be done to `src/runtime/process-worker.ts`

**Comment by kricsleo** [comment]
Sorry, my bad, it's been added now :）

File: src/runtime/process-worker.ts [hunk 1]
```diff
@@ -21,9 +20,11 @@ export default class ProcessWorker implements TinypoolWorker {
   waitForExit!: Promise<void>
 
   initialize(options: Parameters<TinypoolWorker['initialize']>[0]) {
-    const __dirname = dirname(fileURLToPath(import.meta.url))
-
-    this.process = fork(resolve(__dirname, './entry/process.js'), options)
+    // fix https://github.com/tinylibs/tinypool/issues/71
```
Comment (AriPerkkio) [inline src/runtime/process-worker.ts:23]: This will be available in git history.

```suggestion
```

File: src/runtime/thread-worker.ts [hunk 2]
```diff
@@ -10,9 +9,11 @@ export default class ThreadWorker implements TinypoolWorker {
   threadId!: number
 
   initialize(options: Parameters<TinypoolWorker['initialize']>[0]) {
-    const __dirname = dirname(fileURLToPath(import.meta.url))
-
-    this.thread = new Worker(resolve(__dirname, './entry/worker.js'), options)
+    // fix https://github.com/tinylibs/tinypool/issues/71
```
Comment (AriPerkkio) [inline src/runtime/thread-worker.ts:12]: This will be available in git history.

```suggestion
```

**Review by AriPerkkio — APPROVED** [review]

**Review by Aslemammad — APPROVED** [review]
LGTM!

**Pull request merged**